### PR TITLE
Feature/4919 ppa collect data

### PR DESCRIPTION
**Pull request by naveeddotio**
## Description
Mechanism for the collection of ExposureRiskMetadata and UserMetaData

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4919

## Screenshots
N/A

### CODE REVIEW

**Comment by pascal-brause** [comment]
Is that only the PR for storing the data? Because the real collection of the data is missing, isn't it?

File: src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct UserMetadata: Codable {
+	var federalState: String
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift:8]: Wrong data type. I need for submission the correct `federalState`-case from the `PPAFederalState`-enum.

File: src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	var administrativeUnit: String
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift:9]: Wrong data type. I need for submission the correct `Int32` identifier of the json where all units are listed. This is the `districtId`.

**Review by pascal-brause — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift:8]: Perhaps use this?
https://github.com/corona-warn-app/cwa-app-ios/blob/b3e289caad1d38a556f46efd92c17618e32a43ba/src/xcode/ENA/ENA/Source/Scenes/Datadonation/District.swift

File: src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift (hunk 2 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift:9]: Perhaps use this?
https://github.com/corona-warn-app/cwa-app-ios/blob/b3e289caad1d38a556f46efd92c17618e32a43ba/src/xcode/ENA/ENA/Source/Scenes/Datadonation/District.swift

Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift]: Changed it to Int as of districtId, thanks for the hint.

File: src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Models/User/UserMetadata.swift]: Changed it to FederalStateName as discussed

**Comment by naveeddotio** [comment]
> Is that only the PR for storing the data? Because the real collection of the data is missing, isn't it?

We have two type of data collection:

1) For User Data, the data will be taken from UI fields which is part of the other story. 
2) For RiskExposure, we already have the data in RiskCalculationResult, we just need to store that information before submitting it.

**Comment by marcussc** [comment]
> > Is that only the PR for storing the data? Because the real collection of the data is missing, isn't it?
> 
> We have two type of data collection:
> 
> 1. For User Data, the data will be taken from UI fields which is part of the other story.
> 2. For RiskExposure, we already have the data in RiskCalculationResult, we just need to store that information before submitting it.

@naveeddotio  Which other Story?  My understanding was that this story should collect all the MVP data.

**Comment by naveeddotio** [comment]
> > > Is that only the PR for storing the data? Because the real collection of the data is missing, isn't it?
> > 
> > 
> > We have two type of data collection:
> > 
> > 1. For User Data, the data will be taken from UI fields which is part of the other story.
> > 2. For RiskExposure, we already have the data in RiskCalculationResult, we just need to store that information before submitting it.
> 
> @naveeddotio Which other Story? My understanding was that this story should collect all the MVP data.

4788-Privacy-preserving Analytics: Onboarding & Settings, task 4996 datadonation screen

**Comment by naveeddotio** [comment]
> > > Is that only the PR for storing the data? Because the real collection of the data is missing, isn't it?
> > 
> > 
> > We have two type of data collection:
> > 
> > 1. For User Data, the data will be taken from UI fields which is part of the other story.
> > 2. For RiskExposure, we already have the data in RiskCalculationResult, we just need to store that information before submitting it.
> 
> @naveeddotio Which other Story? My understanding was that this story should collect all the MVP data.

4788-Privacy-preserving Analytics: Onboarding & Settings, task 4996 datadonation screen

**Pull request closed without merging**